### PR TITLE
V3/controllers ti bugfix

### PR DESCRIPTION
**Pull request by leefrank9527**
1. Fixed pom.xml about build bugs.
2. Fixed application issues related to TargetInstance and Group.

### CODE REVIEW

File: src/main/java/org/webcurator/core/harvester/agent/HarvestAgentClient.java [hunk 1]
```diff
@@ -237,11 +231,13 @@ public void purgeAbortedTargetInstances(List<String> targetInstanceNames) {
     }
 
     public boolean isValidProfile(String profile) {
+        HttpEntity<String> request = this.createHttpRequestEntity(profile);
```
Comment (hannakoppelaar) [inline src/main/java/org/webcurator/core/harvester/agent/HarvestAgentClient.java]: This json-ifies the XML string, so it is no longer well-formed XML and will be rejected by Heritrix. This is probably easiest to fix by simply doing `HttpEntity<String> request = new HttpEntity<String>(profile);`

File: build.gradle [hunk 2]
```diff
@@ -7,16 +7,14 @@ repositories {
...
-
-    maven {
         url = 'https://repo.maven.apache.org/maven2'
     }
 }
 
 dependencies {
-    compile('org.archive:heritrix:1.14.2-webcuratortool-2.0.2') {
-       exclude group: 'javax.servlet', module: 'servlet-api'
+    compile('org.archive:heritrix:1.14.2-webcuratortool-2.0.1') {
```
[the remark this answers is not in the document]
Comment (leefrank9527) [reply build.gradle]: I don't know what exactlly happened. With 2.0.2 it's failed to harvest, so I switch it to the same lib as WCT2.0 and it does work. If 2.0.1 is the answer maybe we should put it into "webcurator-legacy-lib-dependencies"?

File: build.gradle [hunk 3]
```diff
@@ -7,16 +7,14 @@ repositories {
     mavenLocal()
     maven {
         url = 'https://repo.spring.io/plugins-release/'
-    }
```
[the remark this answers is not in the document]
Comment (leefrank9527) [reply build.gradle]: I got the point. Thank you Hanna. I would recover it to the previous style.

File: src/main/java/org/webcurator/core/harvester/agent/HarvestAgentClient.java (hunk 1 above)
Comment (leefrank9527) [reply src/main/java/org/webcurator/core/harvester/agent/HarvestAgentClient.java]: Accepted and fixed as advised.

**Comment by leefrank9527** [comment]
@hannakoppelaar @obrienben I fixed some issues according to your advices and pushed to this branch again.

File: build.gradle (hunk 2 above)
[the remark this answers is not in the document]
Comment (hannakoppelaar) [reply build.gradle]: Apparently some compatibility issues were encountered between openwayback-core and heritrix which were fixed in the 2.0.2 version of this heritrix fork. I don't know what those issues were. 

What problem did you encounter while harvesting with the 2.0.2 version?

[the remark this answers is not in the document]
Comment (leefrank9527) [reply build.gradle]: Sounds reasonable. I would do more try about 2.0.2.

[the remark this answers is not in the document]
Comment (leefrank9527) [reply build.gradle]: @hannakoppelaar @obrienben  I changed the dependency to org.archive:heritrix:1.14.2-webcuratortool-2.0.2 and it does work now. Refreshed to this PR. ~~

**Pull request merged**